### PR TITLE
issue #3520 Properly show ISO-2022-JP encoded content

### LINKED ISSUE #3520: iso-2022-jp signed message garbled
ref: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgxwLtGlbzXGTcSBkfbffLLgFwttB

It appears we assume UTF-8,  most explicitly in the `fromUtfStr ` method here:

https://github.com/FlowCrypt/flowcrypt-browser/blob/53ce61e2228a55bc624a09d2f354e6f0be90aaea/extension/chrome/elements/pgp_block.ts#L58

But perhaps in other ways as well. 

This may not seem particularly important given the dominance of Unicode, even in Japanese writing. But email is an exception:

> ISO-2022-JP is the 7-bit JIS encoding, that e-mails are usually sent in. It is rarely used in other contexts.

Source: https://www.sljfaq.org/afaq/encodings.html#encodings-Overview-of-the-encoding-schemes

So by not being compatible with ISO-2022-JP, FlowCrypt is incompatible with the dominant form of encoding used for email in Japan!

### PR DESCRIPTION
**Pull request by IvanPizhenko**
This PR fixes showing ISO-2022-JP encoded content

close #3520

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by IvanPizhenko** [comment]
@rrrooommmaaa @tomholub Ready for review

**Comment by IvanPizhenko** [comment]
@rrrooommmaaa @tomholub Ready for review

File: extension/lib/emailjs/emailjs-mime-parser.js [hunk 1]
```diff
@@ -810,7 +810,10 @@
             }
 
             // override charset for text nodes
-            this.charset = this.contentType.params.charset = 'utf-8';
+            // issue #3520 - do not override ISO-2022-JP
+            if (this.charset?.toUpperCase() !== 'ISO-2022-JP') {
```
Comment (rrrooommmaaa) [inline extension/lib/emailjs/emailjs-mime-parser.js:814]: I suggest to exclude based on `Content-Transfer-Encoding: 7bit`
As I mentioned before, 7bit is guaranteed to be not compatible with UTF-8.

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]

File: extension/lib/emailjs/emailjs-mime-parser.js (hunk 1 above)
Comment (rrrooommmaaa) [reply extension/lib/emailjs/emailjs-mime-parser.js:814]: However, this header may be missing...
So we'd better check that either 7bit or it starts with (ISO-2022-JP...) (there is also ISO-2022-JP-2)

Comment (IvanPizhenko) [reply extension/lib/emailjs/emailjs-mime-parser.js:814]: @rrrooommmaaa How do I get value of `Content-Transfer-Encoding`?

**Comment by IvanPizhenko** [comment]
@rrrooommmaaa Please recheck

File: extension/lib/emailjs/emailjs-mime-parser.js [hunk 2]
```diff
@@ -810,10 +810,13 @@
             }
 
             // override charset for text nodes
-            this.charset = this.contentType.params.charset = 'utf-8';
+            // issue #3520 - do not override 7bit econdings and ISO-2022-JPxxx
+            if (!((this.headers['Content-Transfer-Encoding'] && this.headers['Content-Transfer-Encoding'].toLowerCase() === '7bit')
+                || (this.charset && this.charset.toUpperCase().startsWith('ISO-2022-JP')))) {
+                this.charset = this.contentType.params.charset = 'utf-8';
+            }
```
Comment (rrrooommmaaa) [inline extension/lib/emailjs/emailjs-mime-parser.js:817]: have you checked if this method was run and the property `contentTransferEncoding` was set?
image
Also, headers are probably keyed lower-cased

**Comment by IvanPizhenko** [comment]
@rrrooommmaaa Please recheck

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**